### DISCUSSION TITLE
Is it possible to use Step-CA as just an ACME server

### DISCUSSION THREAD

**Discussion by Al7777** [smallstep/certificates#884] [Q&A]
Hello all,

I'm Alex , beginner in ACME in order to implement private ACME CA of my own with Step-CA, Certbot, my own CA module/instance i developed which means that my own CA issues SSL certificate as i want .

That means Step-CA works as just an ACME server while my CA module works certificate issuance/renewal/revokation of SSL certificate

1.My question is 

1. I hope Step-CA works as an ACME server which means my CA  i developed does certificate
issuance/renewal/revocation. If that is possible, Would you please let me know how integrate Step-CA and my
CA module.
Is there any APIs or something which enables integration with my CA ?

 

I hope you're doing well and kindly help..

Thanks all,,

**Comment by tashian** [comment]
You could integrate in a couple ways:
- Directly through our Go libraries and build your own ACME server from our code (this is what Caddy server does)
- Customize RA mode with your own CA backend

Both of these would require a lot of custom development work, and we don't have documentation for either of these approaches.

**Comment by maraino** [comment]
You're describing a custom RA mode, right now we support RA for another step-ca, or Google CAS. There's also a PR for using Vault as the CA. Custom ones can be created if you implement this interface:
https://github.com/smallstep/certificates/blob/dc9db5fbbad2d722b553beaa49775b88e6342f90/cas/apiv1/services.go#L11-L15

As a reference, you can use the implementations in [this directory](https://github.com/smallstep/certificates/tree/master/cas):
* `softcas` is the default implementation of the above interface, and step-ca acts as a CA.
* `stepcas` is an RA and another step-ca acts as a CA.
* `cloudcas` is an RA and Google CAS acts as the CA.
